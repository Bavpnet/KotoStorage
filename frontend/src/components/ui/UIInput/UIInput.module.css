.inp {
  width: 100%;
  float: left;
  padding: 0 0 6px 0;
  background: transparent;
  color: #000;
  font-family: 'Graphik', sans-serif;
  font-size: 1rem;
  font-weight: 400;
  border: none;
  border-bottom: 2px solid #000;
  padding: 0 0 15px 9px;
}

.inp::placeholder {
  color: #000;
  opacity: 0.5;
}

.inp:focus {
  outline: none;
}
