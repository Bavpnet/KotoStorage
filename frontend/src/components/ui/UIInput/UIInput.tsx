import styles from './UIInput.module.css'

export const UIInput: React.FC<
  React.InputHTMLAttributes<HTMLInputElement>
> = props => <input className={styles.inp} {...props} />
