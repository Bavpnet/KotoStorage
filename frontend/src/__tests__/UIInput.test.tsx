import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { UIInput } from '../components/ui/UIInput'

describe('UIInput', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<UIInput />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with placeholder', () => {
    const placeholder = 'test'
    const tree = renderer.create(<UIInput placeholder={placeholder} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with value', () => {
    const { getByRole } = render(<UIInput />)
    const input = getByRole('textbox')

    expect(input).toHaveValue('')

    // simulate user typing
    fireEvent.change(input, { target: { value: 'test' } })

    expect(input).toHaveValue('test')

    // simulate user deleting
    fireEvent.change(input, { target: { value: '' } })

    expect(input).toHaveValue('')
  })
})
