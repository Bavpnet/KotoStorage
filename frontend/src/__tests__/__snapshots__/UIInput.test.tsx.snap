// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIInput renders correctly 1`] = `
<input
  className="inp"
/>
`;

exports[`UIInput renders correctly with placeholder 1`] = `
<input
  className="inp"
  placeholder="test"
/>
`;
